File: C++/프로그래머스/0/120839. 가위 바위 보/가위 바위 보.cpp
#include <string>
#include <vector>

using namespace std;

// switch 문 사용
char get_to_win(char a) {
    switch(a) {
        case '0':
            return '5';
        case '2':
            return '0';
        case '5':
            return '2';
        default:
            return '-1';
    }
    return '-1';
}

string solution(string rsp) {
    string answer = rsp;
    
    for (auto& a: answer) {
        a = get_to_win(a);
    }
    
    return answer;
}

File: Go/프로그래머스/0/120839. 가위 바위 보/가위 바위 보.go
func solution(rsp string) string {
    res := make([]rune, 0, len(rsp))
    
    for _, r := range rsp {
        switch r {
            case '0':
                res = append(res, '5')
            case '2':
                res = append(res, '0')
            case '5':
                res = append(res, '2')
            default:
        }
    }
    return string(res)
}